sys/fmt/fmt.c

```diff
@@ -506,25 +507,24 @@ uint32_t scn_u32_hex(const char *str, size_t n)
     return res;
 }
 
+/* native gets special treatment as native's stdio code is ... special.
+ * And when not building for RIOT, there's no `stdio_write()`.
+ * In those cases, just defer to `printf()`.
+ */
+#if IS_USED(MODULE_STDIO_NATIVE) || !defined(RIOT_VERSION)
```

Comment:
Could you explain to me why this is needed? I fail to understand the reason on my own. E.g. if I remove the guarded code, the unit test still pass for native. What am I misunderstanding?